pkgs/native_assets_builder/test/build_runner/build_runner_test.dart

```diff
@@ -24,6 +24,11 @@ void main() async {
         logger: logger,
       );
 
+      // Make sure the first compile is at least one second after the
+      // package_config.json is written, otherwise dill compilation isn't
+      // cached.
+      await Future<void>.delayed(const Duration(seconds: 1));
```

Comment:
I'd love to. But then we'd also need to mock the file system and the process calls. Then it would be a unit test instead of an integration test. And the test would run blazingly fast.

But I'm worried that by mocking `dart compile kernel` and by mocking `dart my_compiled_hook.dill` instead of running the real hook, we will not actually get the right test coverage.

Reply:
Maybe we should consider splitting our test suite in integration tests and unit tests (like is done in flutter tools), and use the `package:file` and `package:process` for mockability like in flutter tools.